src/rdb.c

```diff
@@ -547,8 +546,7 @@ void *rdbGenericLoadStringObject(rio *rdb, int flags, size_t *lenptr) {
         }
         return buf;
     } else {
-        robj *o = encode ? tryCreateStringObject(SDS_NOINIT,len) :
-                           tryCreateRawStringObject(SDS_NOINIT,len);
```

Comment:
i see this condition was added here: c7822bf38287596a39ea0dfd99654d4680b15d3e
before that it always used to create raw string.
i went looking for it in fear that this PR reverts an intentional change and there was some reason for it.
but i suppose it just just plain mistake.
i.e. at the time, that commit should have changed the code from always creating a raw string into something that automatically decides according to the size, but instead a condition on `encode` was added, which i think was a mistake.